charts/newrelic-infrastructure-v3/templates/agent-configmap.yaml

```diff
@@ -1,4 +1,4 @@
-{{ if .Values.common.agentConfig }}
+{{- if (include "newrelic.compatibility.agentConfig" . ) -}}
```

Comment:
avoiding for the multiple if "(include "newrelic.compatibility.agentConfig" . )" the whole toYaml| FromYaml etc, we can add it, but I see it too verbose

Reply:
I added it in the other cases

Reply:
You are right, copying the whole to yaml from yaml merge would be too verbose.

However perhaps we can get rid of this `if` althogether, and render the configMap unconditionally (empty). Would this be a problem?

Reply:
Actually I prefer the opposite:
 - just for the agent config keep everywhere the simple notation since it is used in many places. In this way we avoid multiple merges, different notations, empty configMaps and complexity. 
   On the down side we are not following the "Overlay pattern" for this config

Reply:
Either way is fine for me, if you think getting rid of the `if` is not the best approach then definitely we cannot overlay it.